### PR TITLE
Fix grad higher order handling TupleVariable

### PR DESCRIPTION
**Pull request by ydwu4**
Previously, we assume the argnums is a **ConstantVariable**. However I accidentally triggered an error on CI where argnums could be a **TupleVariable**. In that case, we have an attribute error when access the .value of argnums. 

This PR adds support for the TupleVariable. It allows the unit test to pass without falling back to eager
"PYTORCH_TEST_WITH_DYNAMO=1 python test/functorch/test_eager_transforms.py -k test_argnums_cpu"

Test Plan:
see modified test.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

**Review by yanboliang — APPROVED** [review]

**Review by anijain2305 — APPROVED** [review]

File: torch/_dynamo/variables/higher_order_ops.py [hunk 1]
```diff
@@ -666,15 +666,37 @@ def call_function(
...
+                    isinstance(var, ConstantVariable) and isinstance(var.value, int)
+                    for var in const_vars
+                ):
+                    raise UserError(
+                        UserErrorType.INVALID_INPUT,
+                        f"argnums is expected to contain int only. Got {const_vars}.",
+                    )
+                return tuple(var.value for var in const_vars)
+
+        indices = to_python_ints(argnums)
```
Comment (kshitij12345) [inline torch/_dynamo/variables/higher_order_ops.py:698]: nit: Can we call these `argnums_v` or something which has `argnums` in the name? I don't think we use `indices` anywhere else in the code or comments.

**Review by kshitij12345 — APPROVED** [review]
Thank you very much for catching and fixing this @ydwu4!

File: torch/_dynamo/variables/higher_order_ops.py (hunk 1 above)
Comment (ydwu4) [reply torch/_dynamo/variables/higher_order_ops.py:698]: Sure!

**Pull request closed without merging**